lib/components/textEditor/richText.js

```diff
@@ -85,13 +87,50 @@ class RichText extends Component {
     });
   }
 
+  // eslint-disable-next-line require-jsdoc
+  getTruncatedContentState(editorState) {
+    const content = stateToHTML(editorState.getCurrentContent());
+    const {
+      maxContentSize: { bytes: maxBytes, warning },
+    } = this.props;
+
+    if (byteCount(content) > maxBytes) {
+      const endTagSize = 10;
+      const blocksFromHTML = convertFromHTML(
+        truncateByBytes(content, maxBytes - endTagSize).replace(/(<|<\/|<\/[a-z]{0,10})$/, ''),
```

Comment:
I don't think we should try and fix anything. We already warn the user, so we do not need to try and make it better. Otherwise this will need tests 😁